src/python.c

```diff
@@ -1183,8 +1183,16 @@ pyobjectToCString(PyObject *pyobject, StringInfo buffer,
 {
     if (pyobject == NULL || pyobject == Py_None)
     {
+        appendBinaryStringInfo(buffer, "NULL", 4);
         return;
     }
+    if (cinfo->atttypoid == JSONOID || cinfo->atttypoid == JSONBOID) {
+       PyObject *p_json = PyImport_ImportModule("json");
```

Comment:
A `timestamp` or `binary` column is exposed as `TIMESTAMP` or `BINARY` postgres, therefore as `timestamp` or `binary` repose, and its value is eventually turned into a JSON string using the convention we choose in our JSON export.

When nested and hidden in a `JSON` column, the same original `timestamp` or `binary` (in Databricks types) turned into strings with some default Python `json` logic, will then render as a string with a `JSON` column, therefore be rendered as an `any` in the query output, where we lost the knowledge it was a `timestamp` or `binary` in the first place. The rendering of `timestamp` or `binary` as strings in Python's JSON doesn't necessarily match how we would have rendered the values in our JSON output.

Reply:
In latest version of the changeset, the code uses a custom method.